### PR TITLE
back end page Content

### PR DESCRIPTION
**Pull request by KotliarovaK**

### CODE REVIEW

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+		['Related Services', '12'],
+		['Get in Touch', '13'],
+		['Related Articles', '14'],
+		['FAQ', '15'],
+	];
+
+	await baseDriverSteps.checkContainerTitlesAndNumbers(containers, expectedData);
+});
+
+test('Check section titles in "Techstack’s Powers\nin Back-End\nDevelopment" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts:63]: pls remove \n from title)

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+	await expect(expertServicesContainer.getByTestId(MainSiteButtons.ScheduleAFreeBackendConsultation)).toHaveText(
+		'Schedule a Free Back-End Consultation'
+	);
+});
+
+test('Check section titles in "Our Tech Arsenal for Back-End Development" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208', async () => {
+	const techArsenalContainer = driver.getByTestId(BackEndDevelopment.TechArsenal);
+	const allSectionTitles = techArsenalContainer.getByTestId(Container.SectionTitle);
+	const testData = ['.NET Stack', 'JVM Stack', 'Node.js Stack', 'Other'];
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts:120]: here you can use
const testData = TechnologyStackData.BackEndTab

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+	const allSectionTitles = caseStudyContainer.getByTestId(Container.SectionTitle);
+	const testData = ['Complexity of\nData Management', 'Integration with\nExisting Systems'];
+
+	await expect(allSectionTitles).toHaveText(testData);
+	await expect(caseStudyContainer.getByTestId(MainSiteImages.BackendCaseStudy)).toBeVisible();
+	await expect(caseStudyContainer.getByTestId(MainSiteButtons.MoreDetails)).toHaveText('More Details');
+});
+
+test('Check member names, roles and CTA in "Our Experts" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208', async () => {
+	const ourExpertsContainer = driver.getByTestId(BackEndDevelopment.OurExperts);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts:136]: in  "Our Leading Back-End Engineers"

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+		'Microservices\nArchitecture',
+		'Back-End Refactoring',
+		'Technical\nBack-End Audit',
+		'API Development\nand Integration',
+	];
+
+	await expect(allSectionTitles).toHaveText(testData);
+});
+
+test('Check section titles, numbers and CTA in "Back-End Development Services" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts:97]: mb in "Back-End Development Services for Any Software Product"

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+
+	expect(actualCarouselIndexesAndTitles).toEqual(expectedCarouselIndexesAndTitles);
+
+	await expect(peekIntoBackendContainer.getByTestId(MainSiteButtons.BoostYourBackend)).toHaveText(
+		'Boost your Back-End'
+	);
+});
+
+test('Check section titles in "Back-End Development Services" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208', async () => {
+	const backendDevelopmentServicesContainer = driver.getByTestId(BackEndDevelopment.BackendDevelopmentServices);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts:242]: mb same with container title

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+});
+
+test('Check carousel section numbers and titles in "A Peek Into Back-End Process" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208', async () => {
+	const peekIntoBackendContainer = driver.getByTestId(BackEndDevelopment.PeekIntoBackend);
+	const carouselSections = await peekIntoBackendContainer.getByTestId(Container.CarouselSection).all();
+	const actualCarouselIndexesAndTitles: Map<string, string> = new Map();
+
+	for (const carouselSection of carouselSections) {
+		const index = await carouselSection.getByTestId(Container.SectionNumber).textContent();
+		const title = await carouselSection.getByTestId(Container.SectionTitle).first().textContent();
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts:215]: can we romeve first() ?

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+		'Cost efficiency',
+		'Customization and flexibility',
+		'Reliability and stability',
+		'Future-proofing',
+	];
+
+	await expect(allSectionTitles).toHaveText(testData);
+});
+
+test('Check carousel section numbers and titles in "A Peek Into Back-End Process" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/backEndDevelopmentContent.test.ts:208]: mb 
Check carousel section numbers and titles**, and CTA** in "A Peek Into Our Back-End Development Process" container...

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

**Comment by KotliarovaK** [comment]
Issues occurred while pushing

**Pull request closed without merging**